pip/qsharp/_qsharp.py

```diff
@@ -245,6 +245,24 @@ def estimate(
         json.loads(get_interpreter().estimate(entry_expr, json.dumps(params)))
     )
 
+def set_quantum_seed(seed: Optional[int]) -> None:
+    """
+    Sets the seed for the quantum random number generator used for measurements.
+
+    :param seed: The seed to use for the quantum random number generator.
+        If None, the seed will be generated from entropy.
+    """
+    get_interpreter().set_quantum_seed(seed)
+
+def set_classical_seed(seed: Optional[int]) -> None:
+    """
+    Sets the seed for the classical random number generator used for standard
+    library random number operations.
+
+    :param seed: The seed to use for the classical random number generator.
+        If None, the seed will be generated from entropy.
+    """
+    get_interpreter().set_classical_seed(seed)
```

Comment:
Will users want to control these separately? To keep the api simple I wonder if we should just expose one `set_random_seed`. 

1. Is the distinction really clear to users which library operations will use the classical rng and which end up using the quantum rng? (Honest question, perhaps this is a concept that's clear to Quantum devs, but I feel like it's pretty opaque)
2. If you're touching this, it's probably because you want your program to be fully deterministic, so I feel like users would just get into the habit of calling _both_ every time to be safe.

Reply:
I really went back and forth debating on that one too. Every time I convinced myself of one side, I'd eventually flip to the other. As you say, it seems very likely that users will always set both because their interest is running the program fully deterministically. But then again, a user who is running many shots of some combined quantum-classical algorithm might want to see the behavior of the quantum with fixed classical randomness, so that makes me want to keep them separate. But in that case, couldn't the user just replace calls to `DrawRandomInt` and/or `DrawRandomDouble` with fixed values? But that is hard to do if they come from a library, which we know we eventually want to support external dependencies... I think I just need someone else to weigh in with a strong preference.

@minestarks and @tcNickolas, do either of you feel strongly one way or the other?

Reply:
In my use cases, classical random numbers affect the structure of the circuit (which gates are applied, with which parameters, etc.), while quantum seed affects the simulation results. 

I'm more interested in the first one, since we don't usually need to reproduce the exact measurement outcomes, while we might want to reproduce the exact circuit we estimate on subsequent runs. (The user doesn't replace calls to DrawRandomInt with fixed values because they're generating 1000 random gates in a for loop, and replacing calls with fixed numbers means effectively generating a hardcoded Q# circuit using non-Q# means, like we did for https://github.com/microsoft/qsharp/blob/main/samples/estimation/estimation-hardcoded-circuit.ipynb - it's possible but not really readable, or even easy to scroll!)

But I can definitely imagine wanting to get different simulation results on each run for that fixed circuit, for example, if I want to gather some statistics about the outcomes. If somebody wants to generate a fixed circuit but then gets deterministic measurement results each time, they might start questioning the fundamental randomness of quantum computing, we don't want that to happen :-)

So, I'd say it's best to keep them separate - it doesn't look like that's too much work (and it's already done anyways :-)), and it's better to give the user more flexibility than less.